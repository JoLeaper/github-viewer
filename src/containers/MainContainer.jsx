import React, { Component } from 'react';
import SearchBar from '../components/SearchBar/SearchBar';
import UserInfo from '../components/UserInfo/UserInfo';
import UserRepos from '../components/UserRepos/UserRepos';
import { getUserInfo } from '../services/getUserInfo';

export default class MainContainer extends Component {
  state = {
    userName: ''
  }

  handleChange = (e) => {
    const newState = {};
    newState[e.target.name] = e.target.value;
    this.setState(newState);
  }
  // componentDidMount() {
  //   getUserInfo('JoLeaper')
  //     .then(res => console.log(res));
  // }

  render() {
    const { userName } = this.state;
    return (
      <div>
        <SearchBar handleChange={this.handleChange}/>
        <UserInfo userName={userName} />
        <UserRepos userName={userName}/>
      </div>
    );
  }
}
