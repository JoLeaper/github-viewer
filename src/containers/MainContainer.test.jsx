import React from 'react';
import { shallow } from 'enzyme';
import MainContainer from './MainContainer';

describe('MainContainer component', () => {
  it('renders MainContainer', () => {
    const wrapper = shallow(<MainContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
