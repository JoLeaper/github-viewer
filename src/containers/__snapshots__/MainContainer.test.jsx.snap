// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer component renders MainContainer 1`] = `
<div>
  <SearchBar
    handleChange={[Function]}
    handleSubmit={[Function]}
  />
  <UserInfo
    userInfo={Object {}}
  />
  <UserRepos
    userReposArray={Array []}
  />
</div>
`;
